### PR TITLE
Bigger OSD grid support

### PR DESCRIPTION
**Pull request by benlumley**
Consider this a draft/proof of concept + not ready for merge, because:

1. The firmware doesn't support this feature yet.....
2. The implementation here should be improved a little - eg: it needs MSP version feature switches; more testing + potentially updating to match however the firmware implements this.

It is configurator side changes to support different grid sizes for the OSD; initially hoping for support along the lines of the HD OSD found in iNav (50 * 18) which is supported by HDZero + others; but the changes make the configurator able relatively easily to support any size.

Not much to it; but in brief I have:

- Added a new 'HD' size alongside the existing PAL/NTSC options.
- Replaced the single width of 30 (front FONT.constants.SIZES.LINE) with new OSD.constants.VIDEO_COLS, matching the existing OSD.constants.VIDEO_LINES
- In CSS, the grid that shows as you drag elements now uses a dashed border instead of a background image, so that we don't need to keep creating new images for every size.
- Also a trivial fix for virtual mode; OSD needs to be passed now - guessing the OSD var was previously in the global scope.

It seems to work in so far as I can use the UI, set up a PAL/NTSC OSD as usual + can also set up an HD one including saving/retrieving it from an FC (running BF 4.3) - positions are saved, survive a power cycle etc.

More than happy to own the feature + work on this further should the firmware team decide to implement the feature.

### CODE REVIEW

File: src/js/tabs/osd.js [hunk 1]
```diff
@@ -1868,13 +1876,13 @@ OSD.updateDisplaySize = function() {
...
     OSD.data.displaySize = {
-        x: FONT.constants.SIZES.LINE,
+        x: OSD.constants.VIDEO_COLS[videoType],
         y: OSD.constants.VIDEO_LINES[videoType],
         total: null,
     };
     // Adjust css background grid
     const previewLayoutElement = $(".tab-osd .display-layout");
-    videoType === 'PAL' ? previewLayoutElement.addClass('video-pal').removeClass('video-ntsc') : previewLayoutElement.addClass('video-ntsc').removeClass('video-pal');
+    videoType === 'PAL' || videoType == 'HD' ? previewLayoutElement.addClass('video-pal').removeClass('video-ntsc') : previewLayoutElement.addClass('video-ntsc').removeClass('video-pal');
```
Comment (haslinghuis) [inline src/js/tabs/osd.js:1885]: Rather would use a new `video-hd` (or maybe `video-hdzero`, `video-walksnail` class) instead of re-using PAL (class).

Comment (benlumley) [reply src/js/tabs/osd.js:1885]: true. don't actually need the class at all any more - it was used to swap the image backgrounds out...

Comment (benlumley) [reply src/js/tabs/osd.js]: removed now.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by nerdCopter** [comment]
line 109, `        LINE: 30,` may be removed.

**Comment by benlumley** [comment]
#3071 supercedes, so closing

**Pull request closed without merging**